src/docs/development/platform-integration/add-to-app-android/add-flutter-screen.md

```diff
@@ -0,0 +1,266 @@
+  // no implementation is necessary.
+}
+```
+
+Second, register your custom `Application` subclass in your `AndroidManifest.xml` file:
+
+```xml
+<application
+  android:name=".MyApplication"
+  // other properties ellided
+  >
+  <!-- content ellided -->
+</application>
+```
+
+Replace `MyApplication` with your `Application` subclass name.
+
+#### Option 3: Manually initialize Flutter
+
+If you'd like to control precisely where and when Flutter is initialized, you can initialize
+Flutter manually. This initialization must occur before you attempt to display any Flutter
+content, or execute any Dart code.
+
+To manually initialize Flutter, execute the following code at the desired location:
+
+```java
+FlutterMain.startInitialization(appContext);
+FlutterMain.ensureInitializationComplete(appContext, null);
+```
+
+The call to `ensureInitializationComplete()` is a blocking call. Make sure you invoke it at a
+place and time that can afford a brief processing pause.
+
+### Step 3: Add FlutterActivity to AndroidManifest.xml
+
+Flutter provides `FlutterActivity` to display a Flutter experience within an Android `Activity`.
+Like any other `Activity`, `FlutterActivity` must be registered in your `AndroidManifest.xml`.
+Add the following XML to your `AndroidManifestxml` file under your `application` tag:
+
+```xml
+<activity
+  android:name=".YourFlutterActivityName"
+  android:launchMode="singleTop"
+  android:theme="@style/LaunchTheme"
+  android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
+  android:hardwareAccelerated="true"
+  android:windowSoftInputMode="adjustResize"
+  />
+```
+
+Replace `YourFlutterActivityName` with the name of your Flutter `Activity`.
+
+### Step 4: Launch FlutterActivity
+
+With `FlutterActivity` registered in your manifest file, add code to launch `FlutterActivity`
+from whatever point in your app that you'd like. The following example shows `FlutterActivity`
+being launched from an `OnClickListener`.
+
+```java
+myButton.addOnClickListener(new OnClickListener() {
+  @Override
+  public void onClick(View v) {
+    startActivity(
+      FlutterActivity.createBuilder().build(currentActivity)
+    );
+  }
+});
+```
+
+The above example assumes that your Dart entrypoint is called `main()`, and your initial
+Flutter route is '/'. If you'd like to change either property, use the `IntentBuilder` to 
+provide your desired properties:
+
+```java
+myButton.addOnClickListener(new OnClickListener() {
+  @Override
+  public void onClick(View v) {
+    startActivity(
+      FlutterActivity
+        .createBuilder()
+        .dartEntrypoint("mySpecialEntrypoint")
+        .initialRoute("/my_route")
+        .build(currentActivity)
+      );
+  }
+});
+```
+
+Replace `"mySpecialEntrypoint"` with the name of your desired Dart entrypoint, and replace
+`"/my_route"` with your desired initial route.
+
+### Step 5: (Optional) Use a cached FlutterEngine
```

Comment:
This is more of a step 4 option 2 than step 5 right?

Reply:
Also, since we're not going to be productionizing the vm+isolate loading time optimization in phase 1, I'd like to flip these options, and recommend a prewarmed engine first and treat the all-in-one solution as a convenience-at-a-cost solution rather than the default solution.

Reply:
I think that when a process is being explained, the initial explanation should cover the process that you get without making any adjustments. If you'd like for me to change the title so that it's not mentioned as a numbered step, then I can do that, but I don't think it's a good idea to jump into engine customization before the reader has been made aware of the fundamental task at hand.

Reply:
There's A: start an engine and B: attach this engine to a view.

I'm asking that we present
1: do A then do B
2: do A+B in one call

as 1 first or 1 and 2 as flat choices rather than 2 first. If this means explaining A and B before offering them choices, that's fine too.

Reply:
I don't think that's how the reader is thinking about the situation. The reader has no knowledge at this point of a `FlutterEngine` or a `FlutterView` or attachment/detachment. At this point the reader is wondering "How do I launch a Flutter experience?" I think we need to answer that simple question before we go into details about engines and caching and implementation details. I would encourage us not to explain complicated things before explaining simple things.

Reply:
Note too though that launching via a pre-warmed engine is our main supported/performant path and we don't want to relegate it to an optional footnote.

Reply:
But it's not a footnote. It's a primary section of the document with as much weight as every other section. Let's chat in person the next time you're in the office. I'm not sure these comments are moving us closer to a resolution.